contracts/0.8.9/WithdrawalQueueERC721.sol

```diff
@@ -146,6 +143,18 @@ contract WithdrawalQueueERC721 is IERC721Metadata, WithdrawalQueue {
         emit NftDescriptorAddressSet(_nftDescriptorAddress);
     }
 
+    /// @notice Finalize requests from last finalized one up to `_lastRequestIdToFinalize`
+    /// @dev ether to finalize all the requests should be calculated using `finalizationValue()` and sent along
+    function finalize(uint256[] calldata _batches, uint256 _maxShareRate) external payable {
```

Comment:
Maybe it's better to leave the `finalize` method in `WithdrawalQueue` but override it here, emitting the event in addition?

Reply:
Idea is great but implementation can be messy because of the payable. And, I'd rather prefer to put all the checks in the top method.